lib/authentication/view/edit_profile_page.dart

```diff
@@ -65,9 +152,87 @@ class _EditProfilePageState extends State<EditProfilePage> {
     );
   }
 
+  AppDialog _changeEmailConfirmationDialog(BuildContext context) {
+    final passwordController = TextEditingController();
+    return AppDialog(
+      title: S.of(context).actionConfirmChangeEmail,
+      message: S.of(context).messageChangeEmail(emailController.text),
+      content: [
+        TextFormField(
+          decoration: InputDecoration(
+            labelText: S.of(context).labelConfirmPassword,
+            hintText: S.of(context).hintPassword,
+          ),
+          obscureText: true,
+          controller: passwordController,
+        ),
+      ],
+      actions: [
+        AppButton(
+          key: ValueKey('change_email_button'),
+          text: S.of(context).actionConfirmChangeEmail,
+          color: Colors.lightBlue,
+          width: 130,
+          onTap: () async {
+            AuthProvider authProvider =
+                Provider.of<AuthProvider>(context, listen: false);
+            if (await authProvider.verifyPassword(
+                password: passwordController.text, context: context)) {
+              if (await authProvider.changeEmail(
+                  email: emailController.text + S.of(context).stringEmailDomain,
+                  context: context)) {
+                Navigator.pop(context, true);
+              } else {
+                Navigator.pop(context, false);
+              }
+            }
+          },
+        )
+      ],
+    );
+  }
+
+  Widget _accountNotVerifiedFooter(BuildContext context) {
```

Comment:
Maybe make this a widget in this file (AccountNotVerifiedFooter) so you can reuse it in the home page card.